subsys/nrf_security/src/drivers/cracen/cracenpsa/src/rsaes_oaep.c

```diff
@@ -212,14 +207,18 @@ int cracen_rsa_oaep_encrypt(const struct sxhashalg *hashalg, struct cracen_rsa_k
 	/* start encoding and request generation of the random seed */
 
 	/* pointer used to walk through the data block DB */
-	uint8_t *datab = workmem.datablockstart;
+	uint8_t *datab = workmem.datablock + digestsz;
 	size_t datablocksz = modulussz - digestsz - 1;
-	size_t paddingstrsz = datablocksz - digestsz - text->sz - 1;
+	size_t paddingstrsz = (modulussz - digestsz - 1) - (digestsz + text->sz + 1);
 
-	datab += paddingstrsz;
+	if (paddingstrsz < 0 || paddingstrsz > datablocksz) {
+		return SX_ERR_TOO_BIG;
+	}
 
 	/* write the padding string PS, consisting of zero octets */
-	safe_memset(datab, wmem_size - (datablockstart_offset + digestsz), 0, paddingstrsz);
+	safe_memset(datab, paddingstrsz, 0, paddingstrsz);
```

Comment:
Nit but why not use the safe_memzero here?

Reply:
Great question. Changed it